index.js

```diff
@@ -2746,6 +2748,59 @@
         return filter(is(type), Just(x));
       });
 
+  //# keys :: StrMap a -> Array String
+  //.
+  //. Returns an array of the keys of the supplied StrMap.
+  //.
+  //. ```javascript
+  //. > S.keys({a: 1, b: 2, c: 3})
+  //. ['a', 'b', 'c']
+  //. ```
+  S.keys =
+  def('keys',
+      {},
+      [$.StrMap(a), $.Array($.String)],
+      function(strMap) { return Object.keys(strMap); });
+
+  //# values :: StrMap a -> Array a
+  //.
+  //. Returns an array of the values of the supplied StrMap.
+  //.
+  //. ```javascript
+  //. > S.values({a: 1, b: 2, c: 3})
+  //. [1, 2, 3]
+  //. ```
+  S.values =
+  def('values',
+      {},
+      [$.StrMap(a), $.Array(a)],
+      function(strMap) {
+        var xs = Object.keys(strMap);
+        for (var idx = 0; idx < xs.length; idx += 1) xs[idx] = strMap[xs[idx]];
+        return xs;
+      });
+
+  //# toPairs :: StrMap a -> Array Pair String a
```

Comment:
We need parens: `Array (Pair String a)`. As written Array appears to be a ternary type. ;)